tao_compiler/mlir/disc/transforms/disc_transform_schedule.cc

```diff
@@ -272,12 +311,65 @@ LogicalResult aarch64GEMMDefaultScheduleFactory(PatternDescription& pd,
 
   Value padForInput = buildGetProducerOfOperand(b, loc, padOp, 0);
   Value padForWeight = buildGetProducerOfOperand(b, loc, padOp, 1);
-  forEachThreadLoop = buildMatchOp(b, loc, variant, {"scf.foreach_thread"});
-  auto outterLoopForN = buildGetParentForOp(b, loc, padForInput, 4);
-  buildCacheRead(b, loc, padForWeight, forEachThreadLoop, {1, 1}, {1, 16}, true,
-                 {2, 0, 1, 3});
-  buildCacheRead(b, loc, padForInput, outterLoopForN, {1, 1}, {6, 1}, true,
-                 {0, 2, 3, 1});
+
+  // Check if we need to pad dimension `m/n/k` if input or weight is packed
+  bool mIsPadded = (M1 != 1) && (M == ShapedType::kDynamicSize ||
+                                 (M > M0 && (M % M0 != 0 || M0 % M1 != 0)) ||
```

Comment:
In what case we'll create a schedule such that `M0 % M1 != 0` ?

Reply:
It's just used to make the check more general. It's often not a good idea to use `M0 % M1 !=0`.